pep.rst

```diff
@@ -519,6 +534,55 @@ Likewise:
 
     mytag()
 
+HTML Example of Rich Return Types
+=================================
+
+Tag functions can be a powerful part of larger processing chains by returning richer objects.
+JavaScript tagged template literals, for example, are not constrained by a requirement to
+return a string. As an example, let's look at an HTML generation system, with a usage and
+"subcomponent":
+
+.. code-block::
+
+    def Menu(icon: str) -> HTML:
+        return html'<img alt="Site Logo" src={icon} />'
+
+    icon = 'acme.png'
+    result = html'<header><{Menu} logo={icon}/></header>'
```

Comment:
I know this is a hypothetical HTML tag syntax, but it reads a little strange to me. Where does the attribute `logo=` get used when invoking `{Menu}`?

Maybe this would read better?:

```python
result = html'<header>{Menu(icon)}</header>`
```

Reply:
I used different names just to point out it an actual symbol in the local namespace, not a magic name, and the caller can have a different name than the callee. If you'd like, I don't mind changing it back.

Reply:
Given that for a given tag function implementation of `html` these two can be equivalent, it might be good to show both.

In either case, `{Menu(icon)}` and `{Menu} logo={icon}` are both resolving the names `Menu` and `icon` within the lexical scope of this tag string.

Reply:
I realize that I didn't really answer @davepeck point. The props-as-args syntax is very familiar to people components over from components in the land of React, Vue, etc. But as you both mention, it doesn't have to be the only way. I will change the example to show that you could choose to pass args inside a call.

@jimbaker Do you envision allowing a mix of HTML attributes and args? E.g. `<{Menu(icon)} class="my-menu" title="My Menu"/>`

Reply:
> @jimbaker Do you envision allowing a mix of HTML attributes and args? E.g. `<{Menu(icon)} class="my-menu" title="My Menu"/>`

I don't know the larger landscape here of props, and how they mix with attrs, but it's certainly possible to allow this mix. Whether that's a good practice or not, I'm not certain.

Reply:
I decided to avoid getting too much into the HTML template language, as this section is about return types. I did though change the example to have another non-interpolated prop:

```python
result = html'<header><{Menu} logo={icon} class="my-menu"/></header>'
img = result.children[0]
assert img.tag == "img"
assert img.props == {"logo": "acme.png", "class": "my-menu"}
```